### DISCUSSION TITLE
What is the relation between `onError` and `ErrorBoundary`?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#369] [Q&A] [answered]
At first I didn't make any error handling, when something occurs, the app stucks.

I found `onError` and `ErrorBoundary` mentioned in the doc, but what is the relation between them and which one should be used in which situation?

**Comment by ryansolid** [comment, marked as answer]
I probably recommend using `ErrorBoundary` if it fits your use case since it handles unmounting of children. It uses `onError` internally but the benefit of `ErrorBoundary` is it ensures children are all resolved up to that point. With `onError` it is possible for children contexts to get passed out of the creator upwards and escape it. `onError` is the lower level primitive and is ok generally higher up in the render tree for logging things but `ErrorBoundary` is more suited for managing the view. I mostly provide `onError` in case someone wishes to implement their own `ErrorBoundary` different than I have created.
